packages/ubuntu_wsl_setup/integration_test/ubuntu_wsl_setup_test.dart

```diff
@@ -0,0 +1,379 @@
+    );
+    await tester.selectOption(
+      label: tester.lang.advancedSetupResolvConfGenerationTitle,
+      value: config?.networkGenerateresolvconf,
+    );
+    await tester.pumpAndSettle();
+
+    await tester.tapButton(label: tester.lang.setupButton, highlighted: true);
+  }
+
+  Future<void> testConfigurationUI(
+    WidgetTester tester, [
+    WSLConfigurationAdvanced? config,
+  ]) async {
+    expectPage(
+        tester, ConfigurationUIPage, (lang) => lang.configurationUITitle);
+
+    await tester.selectOption(
+      label: tester.lang.configurationUILegacyGUIIntegrationTitle,
+      value: config?.interopGuiintegration,
+    );
+    await tester.selectOption(
+      label: tester.lang.configurationUILegacyAudioIntegrationTitle,
+      value: config?.interopAudiointegration,
+    );
+    await tester.selectOption(
+      label: tester.lang.configurationUIAdvancedIPDetectionTitle,
+      value: config?.interopAdvancedipdetection,
+    );
+    await tester.selectOption(
+      label: tester.lang.configurationUIWSLNewsTitle,
+      value: config?.motdWSLnewsenabled,
+    );
+    await tester.selectOption(
+      label: tester.lang.configurationUIAutoMountTitle,
+      value: config?.automountEnabled,
+    );
+    await tester.selectOption(
+      label: tester.lang.configurationUIMountFstabTitle,
+      value: config?.automountMountfstab,
+    );
+    await tester.pumpAndSettle();
+
+    final windowClosed = waitForWindowClosed();
+    await tester.tapButton(label: tester.lang.saveButton, highlighted: true);
+    expect(windowClosed, completion(isTrue));
+  }
+
+  Future<void> testSetupComplete(
+    WidgetTester tester, {
+    String? username,
+  }) async {
+    expectPage(tester, SetupCompletePage, (lang) => lang.setupCompleteTitle);
+
+    if (username != null) {
+      expect(
+        find.text(tester.lang.setupCompleteHeader(username)),
+        findsOneWidget,
+      );
+    }
+
+    final windowClosed = waitForWindowClosed();
+    await tester.tapButton(label: tester.lang.finishButton);
+    expect(windowClosed, completion(isTrue));
+  }
+
+  Future<void> verifyLocale(String locale) async {
+    final path = p.join(
+      Directory.current.path,
+      'subiquity',
+      '.subiquity',
+      'run',
+      'subiquity',
+      'states',
+      'WSLLocale',
+    );
+    expect(File(path).readAsStringSync().trim(), equals('"$locale"'));
+  }
+
+  Future<void> verifyProfile(IdentityData profile) async {
+    // TODO: is the information stored anywhere in dry-run mode?
+  }
+
+  Future<void> verifyIniFile(
+    String fileName,
+    void Function(Config ini) test,
+  ) async {
+    final path = p.join(
+      Directory.current.path,
+      'subiquity',
+      '.subiquity',
+      'etc',
+      fileName,
+    );
+    final lines = File(path).readAsLinesSync();
+    return test(Config.fromStrings(lines));
+  }
+
+  Future<void> verifyBaseConfig(WSLConfigurationBase config) async {
+    await verifyIniFile('wsl.conf', (ini) {
```

Comment:
Note: we are more used to use golden files as best practice where you can pass scenarios (base inputs) and serialize the results in a file, then comparing that file (with a mecanism to refresh them with running the tests with -update). Not a blocker as this is also valid (a little bit less agile), but maybe something to keep in mind? I’m happy to expand my explanations if needed

Reply:
It turns out Flutter exposes the information whether `--update-goldens` was passed, so we can use the "standard" `flutter-test --update-goldens integration_test`, after all. There are instructions in the [README](https://github.com/canonical/ubuntu-desktop-installer/blob/7e4f79fe0ad27aae12439b9244f51c2f9e5f28ca/packages/ubuntu_wsl_setup/integration_test/README.md#update-golden-files).